doc/groups.md

```diff
@@ -1 +1,519 @@
 # Groups Module
+Module | HTTP Method | URL | Description 
+--- | --- | --- | ---
+[Add Groups](#add) | POST | /groups | Add Groups API
+[Edit Groups](#edit) | PUT | /groups/{id} | Edit Groups API
+[View All Groups](#view) | GET | /groups | View Groups API
+[Delete Groups](#delete) | DELETE | /groups/{id} | Delete Groups API
+
+
+### Database
+![](./groups-layer.png)
+
+
+## <a name="add"></a>Add Groups
+
+### Endpoint 
+POST /groups
+
+### Headers
+Key | Value 
+--- | ---
+Content-Type | application/json
+Accept | application/json
+Email | administrator@mail.com
+
+### Request Payloads
+Name | Type | Example Value
+--- | --- | ---
+title | string | Admin  
+description | string | Administrator
+```
+{
+    "title": "Admin",
+    "description": "Administrator"
+}
+
+```
+
+### Response Payloads
+HTTP Code | Status | Description
+--- | --- | ---
+400 | Bad Request | Bad request payload  
+500 | Internal Server Error | some un-handle error in server 
+201 | Created | Created
+```
+{
+    "status_code": "CDC-400",
+    "status_message": "Bad Request",
+    "data": null
+}
+```
+
+```
+{
+    "status_code": "CDC-201",
+    "status_message": "Created",
+    "data": {
+        "id": "1",
```

Comment:
"id": 1,

Reply:
please supply valid type of data. 
"id": "1" is different with "id": 1
"id": "1" is string
"id": 1 is integer

Reply:
ohh, i just knew, thanks sir